website/addons/s3/static/s3NodeConfig.js

```diff
@@ -209,37 +214,101 @@ ViewModel.prototype._importAuthConfirm = function() {
 
 };
 
-ViewModel.prototype.importAuth = function() {
+ViewModel.prototype.connectExistingAccount = function(accountId) {
     var self = this;
-    bootbox.confirm({
-        title: 'Import S3 credentials?',
-        message: 'Are you sure you want to authorize this project with your S3 credentials?',
-        callback: function(confirmed) {
-            if (confirmed) {
-                return self._importAuthConfirm();
+    $osf.putJSON(
```

Comment:
Functions that do AJAX requests should return a promise: 
http://cosdev.readthedocs.org/en/latest/style_guides/javascript.html#promises

Can you also make this change for `ViewModel.prototype.importAuth`?